Author: Arnaud Porterie <arnaud.porterie@docker.com>
Date: Thu, 4 Jun 2015 09:02:14 -0700
Subject: [PATCH] Fix kr/pty import path
Applied-Upstream: https://github.com/docker/docker/pull/13745 (Docker 1.8+)

Index: docker.io/integration-cli/docker_cli_save_load_unix_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_save_load_unix_test.go
+++ docker.io/integration-cli/docker_cli_save_load_unix_test.go
@@ -8,7 +8,7 @@ import (
 	"os"
 	"os/exec"
 
-	"github.com/docker/docker/vendor/src/github.com/kr/pty"
+	"github.com/kr/pty"
 	"github.com/go-check/check"
 )
 
