Description: Minor changes for github.com/opencontainers/runc 0.0.4
Origin: https://github.com/docker/docker/pull/16244
Author: David Calavera <david.calavera@gmail.com>, Tianon Gravi <tianon@debian.org>

diff --git a/daemon/container.go b/daemon/container.go
index 49c1f41..727bd69 100644
--- a/daemon/container.go
+++ b/daemon/container.go
@@ -1088,12 +1088,9 @@ func copyEscapable(dst io.Writer, src io.ReadCloser) (written int64, err error)
 
 func (container *Container) networkMounts() []execdriver.Mount {
 	var mounts []execdriver.Mount
-	mode := "Z"
-	if container.hostConfig.NetworkMode.IsContainer() {
-		mode = "z"
-	}
+	shared := container.hostConfig.NetworkMode.IsContainer()
 	if container.ResolvConfPath != "" {
-		label.Relabel(container.ResolvConfPath, container.MountLabel, mode)
+		label.Relabel(container.ResolvConfPath, container.MountLabel, shared)
 		mounts = append(mounts, execdriver.Mount{
 			Source:      container.ResolvConfPath,
 			Destination: "/etc/resolv.conf",
@@ -1102,7 +1099,7 @@ func (container *Container) networkMounts() []execdriver.Mount {
 		})
 	}
 	if container.HostnamePath != "" {
-		label.Relabel(container.HostnamePath, container.MountLabel, mode)
+		label.Relabel(container.HostnamePath, container.MountLabel, shared)
 		mounts = append(mounts, execdriver.Mount{
 			Source:      container.HostnamePath,
 			Destination: "/etc/hostname",
@@ -1111,7 +1108,7 @@ func (container *Container) networkMounts() []execdriver.Mount {
 		})
 	}
 	if container.HostsPath != "" {
-		label.Relabel(container.HostsPath, container.MountLabel, mode)
+		label.Relabel(container.HostsPath, container.MountLabel, shared)
 		mounts = append(mounts, execdriver.Mount{
 			Source:      container.HostsPath,
 			Destination: "/etc/hosts",
diff --git a/daemon/create.go b/daemon/create.go
index a4a740f..655bd8f 100644
--- a/daemon/create.go
+++ b/daemon/create.go
@@ -123,7 +123,7 @@ func (daemon *Daemon) Create(config *runconfig.Config, hostConfig *runconfig.Hos
 		if err != nil {
 			return nil, nil, err
 		}
-		if err := label.Relabel(v.Path(), container.MountLabel, "z"); err != nil {
+		if err := label.Relabel(v.Path(), container.MountLabel, true); err != nil {
 			return nil, nil, err
 		}
 
diff --git a/daemon/volumes.go b/daemon/volumes.go
index 556e304..ba2b017 100644
--- a/daemon/volumes.go
+++ b/daemon/volumes.go
@@ -231,7 +231,8 @@ func (daemon *Daemon) registerMountPoints(container *Container, hostConfig *runc
 			}
 		}
 
-		if err := label.Relabel(bind.Source, container.MountLabel, bind.Relabel); err != nil {
+		shared := label.IsShared(bind.Relabel)
+		if err := label.Relabel(bind.Source, container.MountLabel, shared); err != nil {
 			return err
 		}
 		binds[bind.Destination] = true
