From 33a8ab29ed9e51697772a0642b8d651b9a845532 Mon Sep 17 00:00:00 2001
From: Pierre Carrier <pierre@meteor.com>, Tianon Gravi <tianon@debian.org>
Date: Thu, 24 Mar 2016 16:14:30 -0700
Subject: [PATCH] docker.service: don't limit tasks
Origin: https://github.com/docker/docker/pull/21491

Signed-off-by: Pierre Carrier <pierre@meteor.com>
---
 contrib/init/systemd/docker.service | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/contrib/init/systemd/docker.service b/contrib/init/systemd/docker.service
index 75cb68c..28d2f4e 100644
--- a/contrib/init/systemd/docker.service
+++ b/contrib/init/systemd/docker.service
@@ -14,6 +14,9 @@ MountFlags=slave
 LimitNOFILE=1048576
 LimitNPROC=1048576
 LimitCORE=infinity
+# Uncomment TasksMax if your systemd version supports it.
+# Only systemd 226 and above support this version.
+TasksMax=infinity
 TimeoutStartSec=0
 # set delegate yes so that systemd does not reset the cgroups of docker containers
 Delegate=yes
