Index: docker.io/integration-cli/check_test.go
===================================================================
--- docker.io.orig/integration-cli/check_test.go
+++ docker.io/integration-cli/check_test.go
@@ -5,7 +5,7 @@ import (
 	"testing"
 	"time"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func Test(t *testing.T) {
Index: docker.io/integration-cli/docker_api_attach_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_api_attach_test.go
+++ docker.io/integration-cli/docker_api_attach_test.go
@@ -7,7 +7,7 @@ import (
 	"strings"
 	"time"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 
 	"golang.org/x/net/websocket"
 )
Index: docker.io/integration-cli/docker_api_containers_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_api_containers_test.go
+++ docker.io/integration-cli/docker_api_containers_test.go
@@ -16,7 +16,7 @@ import (
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/pkg/stringid"
 	"github.com/docker/docker/runconfig"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestContainerApiGetAll(c *check.C) {
Index: docker.io/integration-cli/docker_api_exec_resize_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_api_exec_resize_test.go
+++ docker.io/integration-cli/docker_api_exec_resize_test.go
@@ -5,7 +5,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestExecResizeApiHeightWidthNoInt(c *check.C) {
Index: docker.io/integration-cli/docker_api_exec_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_api_exec_test.go
+++ docker.io/integration-cli/docker_api_exec_test.go
@@ -8,7 +8,7 @@ import (
 	"net/http"
 	"os/exec"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // Regression test for #9414
Index: docker.io/integration-cli/docker_api_images_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_api_images_test.go
+++ docker.io/integration-cli/docker_api_images_test.go
@@ -8,7 +8,7 @@ import (
 	"strings"
 
 	"github.com/docker/docker/api/types"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestApiImagesFilter(c *check.C) {
Index: docker.io/integration-cli/docker_api_info_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_api_info_test.go
+++ docker.io/integration-cli/docker_api_info_test.go
@@ -4,7 +4,7 @@ import (
 	"net/http"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestInfoApi(c *check.C) {
Index: docker.io/integration-cli/docker_api_inspect_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_api_inspect_test.go
+++ docker.io/integration-cli/docker_api_inspect_test.go
@@ -6,7 +6,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestInspectApiContainerResponse(c *check.C) {
Index: docker.io/integration-cli/docker_api_logs_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_api_logs_test.go
+++ docker.io/integration-cli/docker_api_logs_test.go
@@ -9,7 +9,7 @@ import (
 	"strings"
 	"time"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestLogsApiWithStdout(c *check.C) {
Index: docker.io/integration-cli/docker_api_resize_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_api_resize_test.go
+++ docker.io/integration-cli/docker_api_resize_test.go
@@ -5,7 +5,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestResizeApiResponse(c *check.C) {
Index: docker.io/integration-cli/docker_api_stats_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_api_stats_test.go
+++ docker.io/integration-cli/docker_api_stats_test.go
@@ -8,7 +8,7 @@ import (
 	"strings"
 
 	"github.com/docker/docker/api/types"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestCliStatsNoStreamGetCpu(c *check.C) {
Index: docker.io/integration-cli/docker_api_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_api_test.go
+++ docker.io/integration-cli/docker_api_test.go
@@ -8,7 +8,7 @@ import (
 	"time"
 
 	"github.com/docker/docker/api"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestApiOptionsRoute(c *check.C) {
Index: docker.io/integration-cli/docker_api_version_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_api_version_test.go
+++ docker.io/integration-cli/docker_api_version_test.go
@@ -6,7 +6,7 @@ import (
 
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/autogen/dockerversion"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestGetVersion(c *check.C) {
Index: docker.io/integration-cli/docker_cli_attach_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_attach_test.go
+++ docker.io/integration-cli/docker_cli_attach_test.go
@@ -9,7 +9,7 @@ import (
 	"sync"
 	"time"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 const attachWait = 5 * time.Second
Index: docker.io/integration-cli/docker_cli_attach_unix_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_attach_unix_test.go
+++ docker.io/integration-cli/docker_cli_attach_unix_test.go
@@ -9,7 +9,7 @@ import (
 	"time"
 
 	"github.com/docker/docker/pkg/stringid"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 	"github.com/kr/pty"
 )
 
Index: docker.io/integration-cli/docker_cli_build_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_build_test.go
+++ docker.io/integration-cli/docker_cli_build_test.go
@@ -21,7 +21,7 @@ import (
 	"github.com/docker/docker/builder/command"
 	"github.com/docker/docker/pkg/archive"
 	"github.com/docker/docker/pkg/stringutils"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestBuildJSONEmptyRun(c *check.C) {
Index: docker.io/integration-cli/docker_cli_by_digest_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_by_digest_test.go
+++ docker.io/integration-cli/docker_cli_by_digest_test.go
@@ -7,7 +7,7 @@ import (
 	"strings"
 
 	"github.com/docker/docker/utils"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 var (
Index: docker.io/integration-cli/docker_cli_commit_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_commit_test.go
+++ docker.io/integration-cli/docker_cli_commit_test.go
@@ -4,7 +4,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestCommitAfterContainerIsDone(c *check.C) {
Index: docker.io/integration-cli/docker_cli_config_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_config_test.go
+++ docker.io/integration-cli/docker_cli_config_test.go
@@ -9,7 +9,7 @@ import (
 	"path/filepath"
 
 	"github.com/docker/docker/pkg/homedir"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestConfigHttpHeader(c *check.C) {
Index: docker.io/integration-cli/docker_cli_cp_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_cp_test.go
+++ docker.io/integration-cli/docker_cli_cp_test.go
@@ -10,7 +10,7 @@ import (
 	"path/filepath"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 const (
Index: docker.io/integration-cli/docker_cli_create_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_create_test.go
+++ docker.io/integration-cli/docker_cli_create_test.go
@@ -10,7 +10,7 @@ import (
 	"time"
 
 	"github.com/docker/docker/nat"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // Make sure we can create a simple container with some args
Index: docker.io/integration-cli/docker_cli_daemon_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_daemon_test.go
+++ docker.io/integration-cli/docker_cli_daemon_test.go
@@ -17,7 +17,7 @@ import (
 
 	"github.com/docker/libnetwork/iptables"
 	"github.com/docker/libtrust"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerDaemonSuite) TestDaemonRestartWithRunningContainersPorts(c *check.C) {
Index: docker.io/integration-cli/docker_cli_diff_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_diff_test.go
+++ docker.io/integration-cli/docker_cli_diff_test.go
@@ -4,7 +4,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // ensure that an added file shows up in docker diff
Index: docker.io/integration-cli/docker_cli_events_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_events_test.go
+++ docker.io/integration-cli/docker_cli_events_test.go
@@ -10,7 +10,7 @@ import (
 	"sync"
 	"time"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestEventsTimestampFormats(c *check.C) {
Index: docker.io/integration-cli/docker_cli_events_unix_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_events_unix_test.go
+++ docker.io/integration-cli/docker_cli_events_unix_test.go
@@ -10,7 +10,7 @@ import (
 	"os/exec"
 	"unicode"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 	"github.com/kr/pty"
 )
 
Index: docker.io/integration-cli/docker_cli_exec_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_exec_test.go
+++ docker.io/integration-cli/docker_cli_exec_test.go
@@ -14,7 +14,7 @@ import (
 	"sync"
 	"time"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestExec(c *check.C) {
Index: docker.io/integration-cli/docker_cli_exec_unix_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_exec_unix_test.go
+++ docker.io/integration-cli/docker_cli_exec_unix_test.go
@@ -9,7 +9,7 @@ import (
 	"strings"
 	"time"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 	"github.com/kr/pty"
 )
 
Index: docker.io/integration-cli/docker_cli_experimental_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_experimental_test.go
+++ docker.io/integration-cli/docker_cli_experimental_test.go
@@ -6,7 +6,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestExperimentalVersion(c *check.C) {
Index: docker.io/integration-cli/docker_cli_export_import_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_export_import_test.go
+++ docker.io/integration-cli/docker_cli_export_import_test.go
@@ -5,7 +5,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // export an image and try to import it into a new one
Index: docker.io/integration-cli/docker_cli_help_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_help_test.go
+++ docker.io/integration-cli/docker_cli_help_test.go
@@ -8,7 +8,7 @@ import (
 	"unicode"
 
 	"github.com/docker/docker/pkg/homedir"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestHelpTextVerify(c *check.C) {
Index: docker.io/integration-cli/docker_cli_history_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_history_test.go
+++ docker.io/integration-cli/docker_cli_history_test.go
@@ -7,7 +7,7 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // This is a heisen-test.  Because the created timestamp of images and the behavior of
Index: docker.io/integration-cli/docker_cli_images_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_images_test.go
+++ docker.io/integration-cli/docker_cli_images_test.go
@@ -9,7 +9,7 @@ import (
 	"time"
 
 	"github.com/docker/docker/pkg/stringid"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestImagesEnsureImageIsListed(c *check.C) {
Index: docker.io/integration-cli/docker_cli_import_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_import_test.go
+++ docker.io/integration-cli/docker_cli_import_test.go
@@ -4,7 +4,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestImportDisplay(c *check.C) {
Index: docker.io/integration-cli/docker_cli_info_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_info_test.go
+++ docker.io/integration-cli/docker_cli_info_test.go
@@ -5,7 +5,7 @@ import (
 	"strings"
 
 	"github.com/docker/docker/utils"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // ensure docker info succeeds
Index: docker.io/integration-cli/docker_cli_inspect_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_inspect_test.go
+++ docker.io/integration-cli/docker_cli_inspect_test.go
@@ -6,7 +6,7 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestInspectImage(c *check.C) {
Index: docker.io/integration-cli/docker_cli_kill_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_kill_test.go
+++ docker.io/integration-cli/docker_cli_kill_test.go
@@ -4,7 +4,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestKillContainer(c *check.C) {
Index: docker.io/integration-cli/docker_cli_links_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_links_test.go
+++ docker.io/integration-cli/docker_cli_links_test.go
@@ -10,7 +10,7 @@ import (
 	"strings"
 	"time"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestLinksEtcHostsRegularFile(c *check.C) {
Index: docker.io/integration-cli/docker_cli_login_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_login_test.go
+++ docker.io/integration-cli/docker_cli_login_test.go
@@ -4,7 +4,7 @@ import (
 	"bytes"
 	"os/exec"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestLoginWithoutTTY(c *check.C) {
Index: docker.io/integration-cli/docker_cli_logs_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_logs_test.go
+++ docker.io/integration-cli/docker_cli_logs_test.go
@@ -11,7 +11,7 @@ import (
 	"time"
 
 	"github.com/docker/docker/pkg/timeutils"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // This used to work, it test a log of PageSize-1 (gh#4851)
Index: docker.io/integration-cli/docker_cli_nat_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_nat_test.go
+++ docker.io/integration-cli/docker_cli_nat_test.go
@@ -7,7 +7,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func startServerContainer(c *check.C, msg string, port int) string {
Index: docker.io/integration-cli/docker_cli_pause_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_pause_test.go
+++ docker.io/integration-cli/docker_cli_pause_test.go
@@ -5,7 +5,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestPause(c *check.C) {
Index: docker.io/integration-cli/docker_cli_port_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_port_test.go
+++ docker.io/integration-cli/docker_cli_port_test.go
@@ -8,7 +8,7 @@ import (
 	"sort"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestPortList(c *check.C) {
Index: docker.io/integration-cli/docker_cli_proxy_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_proxy_test.go
+++ docker.io/integration-cli/docker_cli_proxy_test.go
@@ -5,7 +5,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestCliProxyDisableProxyUnixSock(c *check.C) {
Index: docker.io/integration-cli/docker_cli_ps_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_ps_test.go
+++ docker.io/integration-cli/docker_cli_ps_test.go
@@ -8,7 +8,7 @@ import (
 	"strings"
 	"time"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestPsListContainers(c *check.C) {
Index: docker.io/integration-cli/docker_cli_pull_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_pull_test.go
+++ docker.io/integration-cli/docker_cli_pull_test.go
@@ -5,7 +5,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // See issue docker/docker#8141
Index: docker.io/integration-cli/docker_cli_push_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_push_test.go
+++ docker.io/integration-cli/docker_cli_push_test.go
@@ -9,7 +9,7 @@ import (
 	"strings"
 	"time"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // pulling an image from the central registry should work
Index: docker.io/integration-cli/docker_cli_rename_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_rename_test.go
+++ docker.io/integration-cli/docker_cli_rename_test.go
@@ -5,7 +5,7 @@ import (
 	"strings"
 
 	"github.com/docker/docker/pkg/stringid"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestRenameStoppedContainer(c *check.C) {
Index: docker.io/integration-cli/docker_cli_restart_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_restart_test.go
+++ docker.io/integration-cli/docker_cli_restart_test.go
@@ -5,7 +5,7 @@ import (
 	"strings"
 	"time"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestRestartStoppedContainer(c *check.C) {
Index: docker.io/integration-cli/docker_cli_rm_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_rm_test.go
+++ docker.io/integration-cli/docker_cli_rm_test.go
@@ -5,7 +5,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestRmContainerWithRemovedVolume(c *check.C) {
Index: docker.io/integration-cli/docker_cli_rmi_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_rmi_test.go
+++ docker.io/integration-cli/docker_cli_rmi_test.go
@@ -5,7 +5,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestRmiWithContainerFails(c *check.C) {
Index: docker.io/integration-cli/docker_cli_run_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_run_test.go
+++ docker.io/integration-cli/docker_cli_run_test.go
@@ -20,7 +20,7 @@ import (
 
 	"github.com/docker/docker/nat"
 	"github.com/docker/libnetwork/resolvconf"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // "test123" should be printed by docker run
Index: docker.io/integration-cli/docker_cli_run_unix_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_run_unix_test.go
+++ docker.io/integration-cli/docker_cli_run_unix_test.go
@@ -14,7 +14,7 @@ import (
 	"time"
 
 	"github.com/docker/docker/pkg/mount"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 	"github.com/kr/pty"
 )
 
Index: docker.io/integration-cli/docker_cli_save_load_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_save_load_test.go
+++ docker.io/integration-cli/docker_cli_save_load_test.go
@@ -10,7 +10,7 @@ import (
 	"sort"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // save a repo using gz compression and try to load it using stdout
Index: docker.io/integration-cli/docker_cli_save_load_unix_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_save_load_unix_test.go
+++ docker.io/integration-cli/docker_cli_save_load_unix_test.go
@@ -9,7 +9,7 @@ import (
 	"os/exec"
 
 	"github.com/kr/pty"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // save a repo and try to load it using stdout
Index: docker.io/integration-cli/docker_cli_search_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_search_test.go
+++ docker.io/integration-cli/docker_cli_search_test.go
@@ -4,7 +4,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // search for repos named  "registry" on the central registry
Index: docker.io/integration-cli/docker_cli_start_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_start_test.go
+++ docker.io/integration-cli/docker_cli_start_test.go
@@ -6,7 +6,7 @@ import (
 	"strings"
 	"time"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // Regression test for https://github.com/docker/docker/issues/7843
Index: docker.io/integration-cli/docker_cli_start_volume_driver_unix_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_start_volume_driver_unix_test.go
+++ docker.io/integration-cli/docker_cli_start_volume_driver_unix_test.go
@@ -13,7 +13,7 @@ import (
 	"path/filepath"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func init() {
Index: docker.io/integration-cli/docker_cli_stats_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_stats_test.go
+++ docker.io/integration-cli/docker_cli_stats_test.go
@@ -5,7 +5,7 @@ import (
 	"strings"
 	"time"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestCliStatsNoStream(c *check.C) {
Index: docker.io/integration-cli/docker_cli_tag_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_tag_test.go
+++ docker.io/integration-cli/docker_cli_tag_test.go
@@ -5,7 +5,7 @@ import (
 	"strings"
 
 	"github.com/docker/docker/pkg/stringutils"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // tagging a named image in a new unprefixed repo should work
Index: docker.io/integration-cli/docker_cli_top_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_top_test.go
+++ docker.io/integration-cli/docker_cli_top_test.go
@@ -4,7 +4,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 func (s *DockerSuite) TestTopMultipleArgs(c *check.C) {
Index: docker.io/integration-cli/docker_cli_version_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_version_test.go
+++ docker.io/integration-cli/docker_cli_version_test.go
@@ -4,7 +4,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // ensure docker version works
Index: docker.io/integration-cli/docker_cli_wait_test.go
===================================================================
--- docker.io.orig/integration-cli/docker_cli_wait_test.go
+++ docker.io/integration-cli/docker_cli_wait_test.go
@@ -6,7 +6,7 @@ import (
 	"strings"
 	"time"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // non-blocking wait with 0 exit code
Index: docker.io/integration-cli/docker_utils.go
===================================================================
--- docker.io.orig/integration-cli/docker_utils.go
+++ docker.io/integration-cli/docker_utils.go
@@ -23,7 +23,7 @@ import (
 	"github.com/docker/docker/opts"
 	"github.com/docker/docker/pkg/ioutils"
 	"github.com/docker/docker/pkg/stringutils"
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 // Daemon represents a Docker daemon for the testing framework.
Index: docker.io/integration-cli/registry.go
===================================================================
--- docker.io.orig/integration-cli/registry.go
+++ docker.io/integration-cli/registry.go
@@ -8,7 +8,7 @@ import (
 	"os/exec"
 	"path/filepath"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 const v2binary = "registry-v2"
Index: docker.io/integration-cli/requirements.go
===================================================================
--- docker.io.orig/integration-cli/requirements.go
+++ docker.io/integration-cli/requirements.go
@@ -9,7 +9,7 @@ import (
 	"os/exec"
 	"strings"
 
-	"github.com/go-check/check"
+	"gopkg.in/check.v1"
 )
 
 type TestCondition func() bool
