Description: Minor changes for github.com/docker/distribution 2.1.1
Forwarded: not-needed
Author: Tianon Gravi <tianon@debian.org>

Index: pkg/registry/registry.go
===================================================================
--- pkg.orig/registry/registry.go
+++ pkg/registry/registry.go
@@ -185,7 +185,7 @@ func addRequiredHeadersToRedirectedReque
 func shouldV2Fallback(err errcode.Error) bool {
 	logrus.Debugf("v2 error: %T %v", err, err)
 	switch err.Code {
-	case v2.ErrorCodeUnauthorized, v2.ErrorCodeManifestUnknown:
+	case errcode.ErrorCodeUnauthorized, v2.ErrorCodeManifestUnknown:
 		return true
 	}
 	return false
