--- docker.io-1.12.6.orig/runconfig/opts/parse_test.go
+++ docker.io-1.12.6/runconfig/opts/parse_test.go
@@ -559,7 +559,7 @@ func TestParseRestartPolicy(t *testing.T
 		"something":          "invalid restart policy something",
 		"always:2":           "maximum restart count not valid with restart policy of \"always\"",
 		"always:2:3":         "maximum restart count not valid with restart policy of \"always\"",
-		"on-failure:invalid": `strconv.ParseInt: parsing "invalid": invalid syntax`,
+		"on-failure:invalid": `strconv.Atoi: parsing "invalid": invalid syntax`,
 		"on-failure:2:5":     "restart count format is not valid, usage: 'on-failure:N' or 'on-failure'",
 	}
 	valids := map[string]container.RestartPolicy{
