Subject: Fix for kernels that have CONFIG_PROC_KCORE=n
Author: Tianon Gravi <admwiggin@gmail.com>
Forwarded: https://github.com/dotcloud/docker/pull/5673
Last-Update: 2014-05-08

diff --git a/pkg/libcontainer/security/restrict/restrict.go b/pkg/libcontainer/security/restrict/restrict.go
index e1296b1..361d076 100644
--- a/pkg/libcontainer/security/restrict/restrict.go
+++ b/pkg/libcontainer/security/restrict/restrict.go
@@ -4,6 +4,7 @@ package restrict
 
 import (
 	"fmt"
+	"os"
 	"syscall"
 
 	"github.com/dotcloud/docker/pkg/system"
@@ -18,8 +19,8 @@ func Restrict(mounts ...string) error {
 			return fmt.Errorf("unable to remount %s readonly: %s", dest, err)
 		}
 	}
-	if err := system.Mount("/dev/null", "/proc/kcore", "", syscall.MS_BIND, ""); err != nil {
-		return fmt.Errorf("unable to bind-mount /dev/null over /proc/kcore")
+	if err := system.Mount("/dev/null", "/proc/kcore", "", syscall.MS_BIND, ""); err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("unable to bind-mount /dev/null over /proc/kcore: %s", err)
 	}
 	return nil
 }
