Description: remove "docker-" prefix on expected "containerd" binaries
Forwarded: not-needed
Author: Tianon Gravi <tianon@debian.org>

diff --git a/integration-cli/docker_test_vars.go b/integration-cli/docker_test_vars.go
index 3559bfdbb..f9dad4809 100644
--- a/integration-cli/docker_test_vars.go
+++ b/integration-cli/docker_test_vars.go
@@ -20,7 +20,7 @@ var (
 	dockerdBinary = "dockerd"
 
 	// path to containerd's ctr binary
-	ctrBinary = "docker-containerd-ctr"
+	ctrBinary = "ctr"
 
 	// the private registry image to use for tests involving the registry
 	registryImageName = "registry"
diff --git a/libcontainerd/remote_unix.go b/libcontainerd/remote_unix.go
index 64a28646b..92a2d49ab 100644
--- a/libcontainerd/remote_unix.go
+++ b/libcontainerd/remote_unix.go
@@ -36,7 +36,7 @@ const (
 	maxConnectionRetryCount      = 3
 	containerdHealthCheckTimeout = 3 * time.Second
 	containerdShutdownTimeout    = 15 * time.Second
-	containerdBinary             = "docker-containerd"
+	containerdBinary             = "containerd"
 	containerdPidFilename        = "docker-containerd.pid"
 	containerdSockFilename       = "docker-containerd.sock"
 	containerdStateDir           = "containerd"
@@ -383,7 +383,7 @@ func (r *remote) runContainerdDaemon() error {
 	if goruntime.GOOS == "solaris" {
 		args = append(args, "--shim", "containerd-shim", "--runtime", "runc")
 	} else {
-		args = append(args, "--shim", "docker-containerd-shim")
+		args = append(args, "--shim", "containerd-shim")
 		if r.runtime != "" {
 			args = append(args, "--runtime")
 			args = append(args, r.runtime)
