Description: allow skipping racy tests with "-test.short"
Author: Tianon Gravi <tianon@debian.org>
Forwarded: no
Bug: https://github.com/docker/docker/issues/22965

diff --git a/pkg/integration/utils_test.go b/pkg/integration/utils_test.go
index d166489..38323ef 100644
--- a/pkg/integration/utils_test.go
+++ b/pkg/integration/utils_test.go
@@ -191,6 +191,10 @@ func TestRunCommandWithOutputAndTimeoutFinished(t *testing.T) {
 }
 
 func TestRunCommandWithOutputAndTimeoutKilled(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping racy test in short mode")
+	}
+
 	// TODO Windows: Port this test
 	if runtime.GOOS == "windows" {
 		t.Skip("Needs porting to Windows")
